core/domain/topic_jobs_one_off.py

```diff
@@ -165,3 +165,40 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class AddCategoryToTopicOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to add category to a topic."""
+    _CATEGORY_ADDED_KEY = 'category_added'
+    _CATEGORY_PRESENT_KEY = 'category_present'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [topic_models.TopicModel]
+
+    @staticmethod
+    def map(item):
+        topic = topic_fetchers.get_topic_by_id(item.id)
+        if hasattr(topic, 'category') and topic.category:
+            yield (AddCategoryToTopicOneOffJob._CATEGORY_PRESENT_KEY, 1)
+            return
+
+        commit_cmds = [topic_domain.TopicChange({
+            'cmd': topic_domain.CMD_UPDATE_TOPIC_PROPERTY,
+            'property_name': 'category',
+            'new_value': 'Mathematics',
```

Comment:
Probably better to use a named constant here and anywhere else you state "Mathematics".

Reply:
Removed.